packages/paste-core/components/anchor/src/index.tsx

```diff
@@ -1,48 +1,53 @@
 import * as React from 'react';
+import * as PropTypes from 'prop-types';
 import {StyledLink} from './styles';
-import {AnchorProps} from './types';
+
+export type AnchorTargets = '_self' | '_blank' | '_parent' | '_top';
+export type AnchorTabIndexes = 0 | -1;
+
+interface Anchor {
+  className?: never;
+  children: NonNullable<React.ReactNode>;
+  href: string;
+  id?: never;
+  onBlur?(event: React.FocusEvent<HTMLElement>): void;
+  onClick?(event: React.MouseEvent<HTMLElement>): void;
+  onFocus?(event: React.FocusEvent<HTMLElement>): void;
+  rel?: string;
+  tabIndex?: AnchorTabIndexes;
+  target?: AnchorTargets;
+}
 
 const EXTERNAL_LINK_REGEX = /^(https?:)[^\s]*$/;
 const EXTERNAL_TARGET_DEFAULT = '_blank';
 const EXTERNAL_REL_DEFAULT = 'noreferrer noopener';
 
 const isExternalUrl = (url: string): boolean => EXTERNAL_LINK_REGEX.test(url);
 
-const handlePropValidation = ({href, tabIndex, children}: AnchorProps): void => {
-  const hasHref = href != null && href !== '';
-  const hasTabIndex = tabIndex != null;
-
-  if (!hasHref) {
-    throw new Error(
-      `[Paste: Anchor] Missing href prop for anchor. Maybe you're looking for the [Paste: Button] component.`
-    );
-  }
-
-  if (children == null) {
-    throw new Error(`[Paste: Anchor] Must have non-null children.`);
-  }
-
-  if (hasTabIndex && !(tabIndex === 0 || tabIndex === -1)) {
-    throw new Error(`[Paste: Anchor] TabIndex must be 0 or -1.`);
-  }
+const Anchor: React.FC<Anchor> = props => (
+  <StyledLink
+    href={props.href}
+    rel={isExternalUrl(props.href) && !props.rel ? EXTERNAL_REL_DEFAULT : props.rel}
+    onBlur={props.onBlur}
+    onClick={props.onClick}
+    onFocus={props.onFocus}
+    tabIndex={props.tabIndex}
+    target={isExternalUrl(props.href) && !props.target ? EXTERNAL_TARGET_DEFAULT : props.target}
```

Comment:
If we don't remove the lint rule, we shouldn't even bother checking `!props.target` here because it's always set.

Reply:
readability nit: `target={props.target || isExternalUrl(props.href) ? EXTERNAL_TARGET_DEFAULT : null}`

Reply:
Going with `target={props.target || isExternalUrl(props.href) ? EXTERNAL_TARGET_DEFAULT : undefined}`

`null` causes a TS error: `Type 'string | null' is not assignable to type 'string | undefined'.`